src/main/java/io/github/lottetreg/echo/Connection.java

```
@@ -1,9 +1,26 @@
 package io.github.lottetreg.echo;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.Socket;
+
 public class Connection {
-  public java.net.Socket socket;
+  public Socket socket;
+
+  Connection() {
+    this.socket = new Socket();
+  }
 
-  Connection(java.net.Socket socket) {
+  public void setSocket(Socket socket) {
     this.socket = socket;
   }
+
+  public InputStream getInputStream() {
+    try {
+      return socket.getInputStream();
+    } catch(IOException e) {
+      System.out.println(e);
+      return null;
```

Generally, I would avoid swallowing an exception and returning `null` if you are going to catch it.  It also requires the callers of the code to do something like this: 
```java 
if (foo != null) {
   doStuff();
}
```
Typically, if I find myself in a situation where I need to handle an exception but I'm not quite sure what I should do with it yet, I throw it up the stack and deal with it at a higher level. Logging is always a good idea too but I know we haven't gotten there yet.  [This blog post](https://www.baeldung.com/java-exceptions) does a decent job talking about exception handling so it might be worth a read.

---

Agreed from me! We were deferring serious exception handling while we paired, but there are enough of these that it's probably time to take this on. A few options:

1) Throw more of these up the call chain and handle them at a high level.
2) Handle them in place if you can do something meaningful when an error occurs. 
3) Kind of like wrapping java.net dependencies like `ServerSocket`, catch the `java.net` exception and throw an exception _you_ own with a meaningful name related to your application. Then do 1) to handle these at a high level.

In this particular case, returning `null` will probably just change your`IOException` in `Connection` into a `NullPointerException` somewhere else. :smile:

---

Thank you both for these suggestions! I'll look into this!